roles/openshift_storage_glusterfs/templates/heketi-endpoints.yml.j2

```diff
@@ -9,7 +9,7 @@ metadata:
 subsets:
 - addresses:
 {% for node in glusterfs_nodes %}
-  - ip: {{ hostvars[node].glusterfs_ip | default(hostvars[node].openshift.common.ip) }}
+  - ip: {{ hostvars[node].glusterfs_ip | default(hostvars[node].openshift.common.ip if "openshift" in hostvars[node] else "") }}
```

Comment:
It's unlikely you want an empty string for an item here.

Reply:
Really I don't want anything, there's no reasonable default. If either of the first two values are missing then it's an error. But we do need to check for `openshift` because it will get evaluated even when it shouldn't.

Reply:
But why? Either way it's malformatted, so the result is the same. Also, what would the conditional even be that wouldn't end up duplicating the the whole variable evaluation?

Reply:
Okay, I think we're saying that we should never get "" and if we do, something else is broken anyway.